core/templates/pages/topic-editor-page/modal-templates/questions-opportunities-select-difficulty-modal.component.ts

```diff
@@ -0,0 +1,109 @@
+import { ImageLocalStorageService } from 'services/image-local-storage.service';
+
+@Component({
+  selector: 'oppia-questions-opportunities-select-difficulty-modal',
+  templateUrl:
+    './questions-opportunities-select-difficulty-modal.component.html'
+})
+export class QuestionsOpportunitiesSelectDifficultyModalComponent
+  extends ConfirmOrCancelModal implements OnInit {
+  @Input() skillId: string;
+  instructionMessage: string;
+  skillIdToRubricsObject: object;
+  skill: Skill;
+  linkedSkillsWithDifficulty: SkillDifficulty[] = [];
+
+  constructor(
+    private alertsService: AlertsService,
+    private assetsBackendApiService: AssetsBackendApiService,
+    private extractImageFilenamesFromModelService:
+      ExtractImageFilenamesFromModelService,
+    private imageLocalStorageService: ImageLocalStorageService,
+    private ngbActiveModal: NgbActiveModal,
+    private skillBackendApiService: SkillBackendApiService
+  ) {
+    super(ngbActiveModal);
+  }
+
+  ngOnInit(): void {
+    this.instructionMessage = (
+      'Select the skill(s) to link the question to:');
+    this.skillBackendApiService.fetchSkillAsync(this.skillId)
+      .then((backendSkillObject) => {
+        this.skill = backendSkillObject.skill;
+        // Skills have SubtitledHtml fields that can contain images. In
+        // order to render them in the contributor dashboard, we parse the
+        // HTML fields in the Skill to get a list of filenames, fetch
+        // these images and store their corresponding base64 URLs in the local
+        // storage. The image components will use the data from the local
+        // storage to render the image.
+        let imageFileFetchPromises = [];
+        let imageFilenames = (
+          this.extractImageFilenamesFromModelService.getImageFilenamesInSkill(
+            this.skill));
+        imageFilenames.forEach(imageFilename => {
+          imageFileFetchPromises.push(this.assetsBackendApiService.loadImage(
+            AppConstants.ENTITY_TYPE.SKILL, this.skillId,
+            imageFilename));
+        });
+        Promise.all(imageFileFetchPromises).then(files => {
+          files.forEach(file => {
+            const reader = new FileReader();
+            reader.onload = () => {
+              const imageData = reader.result as string;
+              this.imageLocalStorageService.saveImage(file.filename, imageData);
+            };
+            reader.readAsDataURL(file.data);
+          });
+          this.linkedSkillsWithDifficulty = [
+            SkillDifficulty.create(
+              this.skillId, this.skill.getDescription(),
+              AppConstants.DEFAULT_SKILL_DIFFICULTY)
+          ];
+          this.skillIdToRubricsObject = {};
+          this.skillIdToRubricsObject[this.skillId] =
+            this.skill.getRubrics();
+        });
+      }, (error) => {
+        this.alertsService.addWarning(
+          `Error populating skill: ${error}.`);
```

Comment:
Please check if the "error" here is a string or a response (we may have to do `response.error.error` or similar, not sure).

Reply:
Here "error" is a string value received. The same way is used at different places in the codebase. (For reference blog-dashboard-page.component.ts line 103)

Reply:
Ah, I see. I initially thought the error callback was for the "Promise.all(imageFileFetchPromises)" on ln 77. Thanks for checking!